src/store/notebook/slice.ts

```diff
@@ -0,0 +1,88 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { initialState } from './types';
+import { getNotebookClusterStatus, startNotebookCluster, stopNotebookCluster } from './thunks';
+import { NotebookApiStatus } from 'services/api/notebook/model';
+
+export const notebookState: initialState = {
+  isLoading: false,
+  url: null,
+  status: NotebookApiStatus.unverified,
+};
+
+const notebookSlice = createSlice({
+  name: 'notebook',
+  initialState: notebookState,
+  reducers: {
+    toggleLoading: (state, action: PayloadAction<boolean>) => ({
+      ...state,
+      isLoading: action.payload,
+    }),
+    reset: () => ({
+      ...notebookState,
+    }),
+    resetError: (state) => ({
+      ...state,
+      error: null,
+    }),
+  },
+  extraReducers: (builder) => {
+    // START cluster
```

Comment:
comments like these // START cluster , in my opinion, bring very little value especially since your function methods are well-named.

Reply:
Nah I like them, it divide all the calling block, easier to read when you have many reducers.